overlord/snapstate/backend/link.go

```diff
@@ -181,6 +181,16 @@ func (b Backend) generateWrappers(s *snap.Info, linkCtx LinkContext) error {
 	}
 	cleanupFuncs = append(cleanupFuncs, wrappers.RemoveSnapBinaries)
 
+	// check DaemonStartup settings and do not start services on first
+	// install that are inhibited
+	if linkCtx.FirstInstall {
+		for _, app := range s.Apps {
+			if app.DaemonStartup == snap.DaemonStartupInhibit {
+				disabledSvcs = append(disabledSvcs, app.Name)
+			}
+		}
+	}
+
```

Comment:
This will conflict with changed semantics in #8892 ; after my changes to wrappers and snapstate land all the units will be created unconditionally (but not enabled) on link, and will get enabled later on start-snap-services. At this point I'd prefer for 8892 to land first and then adjust this code - it will need to be executed from startSnapServices I suppose, but there is no linkCtx there so other adjustments will be required.

Reply:
I agree with Pawel on the ordering here. About the name, something parallel to refresh-mode would be best, install-mode is tempting but we'll see.

Reply:
Happy to wait for 8892 to land first and tweak then

Reply:
The PR to wait for is now https://github.com/snapcore/snapd/pull/8960

Reply:
This is now based on 8960